code/game/machinery/limbgrower.dm

```diff
@@ -0,0 +1,255 @@
+	L.name = "Synthetic [selected_category] Limb"
+	L.desc = "A synthetic [selected_category] limb that will morph on its first use in surgery. This one is for the [selected_part]"
+	L.loc = loc;
+
+/obj/machinery/limbgrower/RefreshParts()
+	reagents.maximum_volume = 0
+	for(var/obj/item/weapon/reagent_containers/glass/G in component_parts)
+		reagents.maximum_volume += G.volume
+		G.reagents.trans_to(src, G.reagents.total_volume)
+	var/T=1.2
+	for(var/obj/item/weapon/stock_parts/manipulator/M in component_parts)
+		T -= M.rating*0.2
+	prod_coeff = min(1,max(0,T)) // Coeff going 1 -> 0,8 -> 0,6 -> 0,4
+
+/obj/machinery/limbgrower/proc/main_win(mob/user)
+	var/dat = "<div class='statusDisplay'><h3>Limb Grower Menu:</h3><br>"
+	dat += "<A href='?src=\ref[src];menu=[LIMBGROWER_CHEMICAL_MENU]'>Chemical Storage</A>"
+	dat += materials_printout()
+	dat += "<table style='width:100%' align='center'><tr>"
+
+	for(var/C in categories)
+		if(C=="special" && !emag)	//Only want to show special when console is emagged
+			continue
+
+		dat += "<td><A href='?src=\ref[src];category=[C];menu=[LIMBGROWER_CATEGORY_MENU]'>[C]</A></td>"
+		dat += "</tr><tr>"
+		//one category per line
+
+	dat += "</tr></table></div>"
+	return dat
+
+/obj/machinery/limbgrower/proc/category_win(mob/user,selected_category)
+	var/dat = "<A href='?src=\ref[src];menu=[LIMBGROWER_MAIN_MENU]'>Return to main menu</A>"
+	dat += "<div class='statusDisplay'><h3>Browsing [selected_category]:</h3><br>"
+	dat += materials_printout()
+
+	for(var/v in files.known_designs)
+		var/datum/design/D = files.known_designs[v]
+		if(!(selected_category in D.category))
+			continue
+		if(disabled || !can_build(D))
+			dat += "<span class='linkOff'>[D.name]</span>"
+		else
+			dat += "<a href='?src=\ref[src];make=[D.id];multiplier=1'>[D.name]</a>"
+		dat += "[get_design_cost(D)]<br>"
+
+	dat += "</div>"
+	return dat
+
+
+/obj/machinery/limbgrower/proc/chemical_win(mob/user)
+	var/dat = "<A href='?src=\ref[src];menu=[LIMBGROWER_MAIN_MENU]'>Return to main menu</A>"
+	dat += "<div class='statusDisplay'><h3>Browsing Chemical Storage:</h3><br>"
+	dat += materials_printout()
+
+	for(var/datum/reagent/R in reagents.reagent_list)
+		dat += "[R.name]: [R.volume]"
+		dat += "<A href='?src=\ref[src];disposeI=[R.id]'>Purge</A><BR>"
+
+	dat += "</div>"
+	return dat
+
+/obj/machinery/limbgrower/proc/materials_printout()
+	var/dat = "<b>Total amount:></b> [reagents.total_volume] / [reagents.maximum_volume] cm<sup>3</sup><br>"
+	return dat
+
+/obj/machinery/limbgrower/proc/can_build(datum/design/D)
+	return (reagents.has_reagent("synthflesh", D.reagents["synthflesh"]*prod_coeff)) //Return whether the machine has enough synthflesh to produce the design
+
+/obj/machinery/limbgrower/proc/get_design_cost(datum/design/D)
+	var/dat
+	if(D.reagents["synthflesh"])
+		dat += "[D.reagents["synthflesh"] * prod_coeff] Synthetic flesh "
+	return dat
+
+/obj/machinery/limbgrower/emag_act(mob/user)
+	if(emag==1)
```

Comment:
use the define

     emag == TRUE

and

    emag = TRUE

It's a little bit clearer that the value is a boolean (sort of) flag